lib/src/map/gestures/map_interactive_viewer.dart

```diff
@@ -0,0 +1,308 @@
+  @override
+  void dispose() {
+    _controller.removeListener(reload);
+    super.dispose();
+  }
+
+  /// Calls [setState] on the [MapInteractiveViewer] widget to refresh the
+  /// widget.
+  void reload() {
+    if (mounted) setState(() {});
+  }
+
+  /// Widget build method
+  @override
+  Widget build(BuildContext context) {
+    final useDoubleTapCallback =
+        _doubleTap != null || _doubleTapDragZoom != null;
+    final useScaleCallback = _keyTriggerDragRotate != null ||
+        _drag != null ||
+        _doubleTapDragZoom != null ||
+        _twoFingerInput != null;
+
+    return Listener(
+      onPointerDown: (event) {
+        _controller.stopAnimationRaw();
+        _options.onPointerDown?.call(
+          event,
+          _camera.offsetToCrs(event.localPosition),
+        );
+      },
+      onPointerHover: _options.onPointerHover == null
+          ? null
+          : (event) => _options.onPointerHover!.call(
+                event,
+                _camera.offsetToCrs(event.localPosition),
+              ),
+      onPointerCancel: _options.onPointerCancel == null
+          ? null
+          : (event) => _options.onPointerCancel!.call(
+                event,
+                _camera.offsetToCrs(event.localPosition),
+              ),
+      onPointerUp: _options.onPointerUp == null
+          ? null
+          : (event) => _options.onPointerUp!.call(
+                event,
+                _camera.offsetToCrs(event.localPosition),
+              ),
+      onPointerSignal: (event) {
+        switch (event) {
+          case final PointerScrollEvent event:
+            // mouse scroll wheel
+            // `stopAnimationRaw()` will probably get moved to the service
+            // to handle animated zooming with the scroll wheel but
+            // we keep it here for now.
+            _controller.stopAnimationRaw();
+            _scrollWheelZoom?.submit(event);
+            break;
+          case final PointerScaleEvent event:
+            // Trackpad pinch gesture, in case the pointerPanZoom event
+            // callbacks can't be used and trackpad scrolling must still use
+            // this old PointerScrollSignal system.
+            //
+            // This is the case if not enough data is
+            // provided to the Flutter engine by platform APIs:
+            // - On **Windows**, where trackpad gesture support is dependent on
+            // the trackpad’s driver,
+            // - On **Web**, where not enough data is provided by browser APIs.
+            //
+            // https://docs.flutter.dev/release/breaking-changes/trackpad-gestures#description-of-change
+            _trackpadLegacyZoom?.submit(event);
+            break;
+        }
+      },
+      // Trackpad gestures on most platforms since flutter 3.3 use
+      // these onPointerPanZoom* callbacks.
+      // See https://docs.flutter.dev/release/breaking-changes/trackpad-gestures
```

Comment:
Do you think we should provide a link to these docs and/or a short explanation in our docs?

Reply:
I brief explaination of every gesture in the docs would probably be useful especially because there are gestures that are not well known (e.g. double tap drag to zoom). 
The linked documentation is more for better understanding when working with the internal code. Therefore either not mentioned in our docs or as "for more information in this see here"-thingy.